pkg/server/cache/entrycache/fullcache.go

```diff
@@ -208,6 +257,47 @@ func (c *FullEntryCache) crawl(parentID spiffeID, seen map[spiffeID]struct{}) []
 	return entries
 }
 
+func (c *FullEntryCache) createOrUpdateEntry(commonEntry *common.RegistrationEntry) error {
+	protoEntry, err := api.RegistrationEntryToProto(commonEntry)
+	if err != nil {
+		return err
+	}
+
+	parentID := spiffeIDFromProto(protoEntry.ParentId)
+	cacheEntries := c.entries[parentID]
+
+	var i int
+	for i = 0; i < len(cacheEntries); i++ {
+		if cacheEntries[i].Id == protoEntry.Id {
+			fmt.Println("Updated entry")
+			cacheEntries[i] = protoEntry
+			break
+		}
+	}
+	if i == len(cacheEntries) {
+		fmt.Println("Created entry")
+		cacheEntries = append(cacheEntries, protoEntry)
+	}
+
+	c.entries[parentID] = cacheEntries
+
+	return nil
+}
+
+func (c *FullEntryCache) deleteEntry(entryID string) {
+	fmt.Println("Deleting entry")
+	for parentID, cacheEntries := range c.entries {
+		for i := 0; i < len(cacheEntries); i++ {
+			if cacheEntries[i].Id == entryID {
+				fmt.Printf("Deleted entry %s", entryID)
+				cacheEntries = append(cacheEntries[:i], cacheEntries[i+1:]...)
+				c.entries[parentID] = cacheEntries
+				return
+			}
+		}
+	}
+}
```

Comment:
@azdagron By only saving the entry ID we have to step through the whole map. What do you think of also adding the parentID to the events table to make this lookup more efficient?

Reply:
@azdagron Any thoughts on this?

Reply:
I don't think can rely on the parent ID in the events table. The parent Id can be mutated so the parent ID reflected in the event and the parent ID on the latest view of the entry can diverge. The in-memory entry cache data structure was not designed for mutability in mind since it was rebuilt in its entirety on every cache reload. If we need to add some additional indices into the cache data structure to aid mutability,  we should do that.

Reply:
Its only really needed for delete operations since there is not latest version of the entry. It would avoid that complex logic above.